File: APM/Java/Interpreter/src/ro/sandorrobertk94/domain/VariableExpression.java
package ro.sandorrobertk94.domain;

/**
 * Created by Robert on 10/15/2015.
 */
public class VariableExpression extends Expression {
    String variableID;

    public VariableExpression(String variableID) {
        this.variableID = variableID;
    }

    public VariableExpression() {
        variableID = null;
    }

    public String getVariableID() {
        return variableID;
    }

    public void setVariableID(String variableID) {
        this.variableID = variableID;
    }

    @Override
    public String toString() {
        return this.variableID;
    }

    @Override
    public Integer eval(MyIDictionary symbolTable) {
        try {
            Object o = symbolTable.lookUp(variableID);
            if (o instanceof Integer) {
                return (Integer) o;
            } else {
                throw new Exception();
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return null;
    }
}

File: APM/CSharp/Interpreter/Interpreter/domain/VariableExpression.cs
using System;

namespace Interpreter.domain
{
    public class VariableExpression : Expression
    {
        public VariableExpression()
        {
            VariableId = null;
        }

        public VariableExpression(string variableId)
        {
            VariableId = variableId;
        }

        public string VariableId { get; set; }

        public override int Eval(MyIDictionary symbolTable)
        {
            var o = symbolTable.LookUp(VariableId);
            if (o is int)
            {
                return (int) o;
            }
            throw new Exception();
        }

        public override string ToString()
        {
            return VariableId;
        }
    }
}